### PR TITLE
Tests refactoring

### PR DESCRIPTION
**Pull request by kiryalovik**

### CODE REVIEW

File: domain/service/noteService.go [hunk 1]
```diff
@@ -4,126 +4,119 @@ package service
...
-	Add(ctx context.Context, barID string, note entity.Note) (entity.Note, error)
-	Insert(ctx context.Context, barID string, position int64, note entity.Note) (entity.Note, error)
-	List(ctx context.Context, barID string) ([]*entity.Note, error)
-	DeleteOne(ctx context.Context, barID string, number int64, position int64) error
-	DeleteMany(ctx context.Context, barIDs []string) error
-	GetSize(ctx context.Context, barID string) (big.Rat, error)
+	Add(ctx context.Context, measureID string, note *entity.Note) error
+	Insert(ctx context.Context, measureID string, position int64, note *entity.Note) error
+	List(ctx context.Context, measureID string) ([]*entity.Note, error)
+	DeleteOne(ctx context.Context, measureID string, number int64, position int64) error
```
Comment (vetheslav) [inline domain/service/noteService.go]: `DeleteOne(ctx context.Context, measureID string, number, position int64) error `

File: domain/service/noteService_test.go [hunk 2]
```diff
@@ -2,226 +2,389 @@ package service
...
-		note      entity.Note
-		wantError bool
+		note      *entity.Note
+	}
+
+	tt := []struct {
+		name     string
+		args     args
+		expected *entity.Note
+		err      string
```
Comment (vetheslav) [inline domain/service/noteService_test.go]: Why don't you use error type?

File: domain/service/noteService.go [hunk 3]
```diff
@@ -4,126 +4,119 @@ package service
...
 	//  	-> if note is a pause
 	//			-> if pause - update pause note with this note
 	//		-> if note values equal
 	//			-> data is out of sync, write to log, do nothing
 	if err != nil {
-		return note, errors.Wrapf(err, insertErrorF, barID, position)
+		return errors.Wrap(err, "can't insert note")
 	}
-	return note, err
+	return err
```
Comment (vetheslav) [inline domain/service/noteService.go]: `return nil`

File: domain/service/noteService.go [hunk 4]
```diff
@@ -4,126 +4,119 @@ package service
...
 	}
 	note.Position = position + 1
-	err = s.repo.Insert(ctx, barID, note)
+	err = s.repo.Insert(ctx, measureID, note)
 	if err != nil {
-		return note, errors.Wrapf(err, addErrorF, barID)
+		return errors.Wrap(err, "can't insert note to repo")
 	}
-	return note, err
+	return err
```
Comment (vetheslav) [inline domain/service/noteService.go]: `return nil`

File: domain/service/noteService_test.go [hunk 5]
```diff
@@ -2,226 +2,389 @@ package service
...
 		position  int64
-		wantError bool
+		note      *entity.Note
+	}
+
+	tt := []struct {
+		name     string
+		args     args
+		expected *entity.Note
+		err      string
```
Comment (vetheslav) [inline domain/service/noteService_test.go]: Why don't use error type?

File: domain/service/noteService_test.go [hunk 6]
```diff
@@ -2,226 +2,389 @@ package service
...
 		measureID string
-		res       []*entity.Note
-		wantError bool
+	}
+
+	tt := []struct {
+		name     string
+		args     args
+		expected []*entity.Note
+		err      string
```
Comment (vetheslav) [inline domain/service/noteService_test.go]: Why don't use error type?

File: domain/service/noteService_test.go [hunk 7]
```diff
@@ -2,226 +2,389 @@ package service
...
-	for testName, testCase := range tests {
-		actual, err := srv.List(context.Background(), testCase.measureID)
-		assert.Exactly(t, actual, testCase.res, fmt.Sprintf(errorF, testName))
-		checkError(t, testName, testCase.wantError, err)
+	for _, tc := range tt {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			actual, err := srv.List(tc.args.ctx, tc.args.measureID)
+			assert.Exactly(t, actual, tc.expected)
+			checkError(t, tc.err, err)
```
Comment (vetheslav) [inline domain/service/noteService_test.go]: you don't need another function. You could use only assert.Equal to check the error

File: infrastructure/datastore/mongo/mongo.go [hunk 8]
```diff
@@ -13,18 +13,20 @@ import (
...
 // Collection creates mongoDB connection and initialize collection.
-func Collection(config datastore.DBConfig) (*mongo.Collection, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
+func Collection(ctx context.Context, config datastore.DBConfig) (*mongo.Collection, error) {
 	mongoURI := fmt.Sprintf("mongodb://%s:%s", config.Host, config.Port)
 	client, err := mongo.Connect(ctx, options.Client().ApplyURI(mongoURI))
 	if err != nil {
-		return nil, errors.Wrap(err, "can't connect to mongoDB server")
+		return nil, errors.Wrap(err, cantConnectError)
```
Comment (vetheslav) [inline infrastructure/datastore/mongo/mongo.go]: I don't think you need to wrap errors in mongo package. But it up to you

File: interface/grpc/mockNoteService_test.go [hunk 9]
```diff
@@ -18,18 +18,18 @@ const (
 
 type mockNoteService struct{}
 
-func (d *mockNoteService) Add(ctx context.Context, measureID string, note entity.Note) (entity.Note, error) {
+func (d *mockNoteService) Add(ctx context.Context, measureID string, note *entity.Note) error {
```
Comment (vetheslav) [inline interface/grpc/mockNoteService_test.go]: Вообще моки немного по другому делаются. Посмотри пример в testify.mock

File: domain/service/noteService_test.go (hunk 2 above)
Comment (Tessen) [reply domain/service/noteService_test.go]: fixed

File: interface/grpc/mockNoteService_test.go (hunk 9 above)
Comment (Tessen) [reply interface/grpc/mockNoteService_test.go]: Переделал под testify.mock

**Pull request merged**